python/cherrypy/config.yaml

```diff
@@ -3,9 +3,8 @@ framework:
   version: 18.3
 
 command: >
-  gunicorn  \
-    --log-level warning  \
-    --bind 0.0.0.0:3000  \
-    --reuse-port  \
-    --workers $(nproc)  \
-      server:app
+  uwsgi \
+    --socket 0.0.0.0:3000 \
```

Comment:
Maybe also add
```suggestion
    --socket 0.0.0.0:3000 \
    --async 100 \
```

Reply:
what does it means ?

Reply:
I'm waiting for this to approve.

Reply:
It makes uswgi have async cores on its side.

Reply:
ah ok, but why `100` ?

Reply:
Maybe it should be the same variable used in other samples $(nproc)

Reply:
No, it's not about threads. AFAIK there's one thread for I/O loop, it's about concurrency, the concept unrelated to CPU cores.

Reply:
https://uwsgi-docs.readthedocs.io/en/latest/Async.html
basically, this number is the maximum concurrent HTTP requests taken by uwsgi

Reply:
I tried out 1024 locally, which seems to be not loading my laptop at all. So maybe some adjustments will be needed.

Reply:
I see ... This is to adjust concurrency level

But :

   + in **production** we **CAN NOT** know in advance concurrency level
   + what problem arrive if I reach up this level ?

Reply:
more, I have `*** DANGER *** async mode without coroutine/greenthread engine loaded !!!` when running with `async` switch

Reply:
That's fine. IRL you have to plan the load for production. I believe, if you reach the concurrency level, the new connections will be reset or kept hanging.

Reply:
on my side, I have :

| async <number> | latency average |
|:---------------------|:-----------------------|
| 1000 | 102.97 ms |
| 100 |  88.91 ms |
| 10 |  82.67 ms |
| 8 (number of my machine cores) | 88.49 ms |
| 6 |  87.23 ms |
| 5 | 86.54 ms |
| 2 |  112.17 ms |
| without | 128 ms |

I do not understand

btw, how is run `CherryPy` on **production** ? purpose `here` is to be close to **production**

Reply:
CherryPy doesn't dictate best practices, it's a low-level minimalistic framework. It's very flexible and it's really up to users to decide how to deploy it.

Reply:
I think that in this case, this implementation need to extract the best from CherryPy.

Reply:
@murilobr @webknjaz I recommand to try with `10`, and we will see

Reply:
I think that this number is quite low but okay. Let's see how it goes. It depends on what load you're going to generate.